### PR TITLE
[web] remove dependency on web_installers and transitively all code that used it

### PR DESCRIPTION
**Pull request by yjbanov**
Remove all dead code related to iOS Simulators and the dependency on `web_installers`.

### CODE REVIEW

File: lib/web_ui/dev/safari_installation.dart [hunk 1]
```diff
@@ -88,3 +86,260 @@ Future<BrowserInstallation> getOrInstallSafari({
...
+    final String phoneInfoAfterId = phoneInfo.substring(endIndexOfPhoneId + 1);
+    final String simulatorStatus = phoneInfoAfterId.substring(
+        phoneInfoAfterId.indexOf('(') + 1, phoneInfoAfterId.indexOf(')'));
+    return IosSimulator._(simulatorStatus == 'Booted', simulatorId);
+  }
+
+  Future<String> _listExistingSimulators(
+      int osMajorVersion, int osMinorVersion) async {
+    final io.ProcessResult versionResult =
+        await io.Process.run('xcrun', <String>['simctl', 'list']);
```
Comment (jmagman) [inline lib/web_ui/dev/safari_installation.dart:258]: I know you're just copying this, but `simctl list` has a `-j` option to get this as json so the text output doesn't need to be parsed.
You can also pass in the version and that you're just looking for "available" devices. On my machine this:
```
xcrun simctl list devices "iOS 12.4" available -j
```
outputs:
```
{
  "devices" : {
    "com.apple.CoreSimulator.SimRuntime.watchOS-8-5" : [

    ],
    "com.apple.CoreSimulator.SimRuntime.iOS-14-0" : [

    ],
    "com.apple.CoreSimulator.SimRuntime.iOS-13-0" : [

    ],
    "com.apple.CoreSimulator.SimRuntime.iOS-12-4" : [
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/5A1AE1B3-BB51-4AA9-8B06-853BF9BDD66D\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/5A1AE1B3-BB51-4AA9-8B06-853BF9BDD66D",
        "udid" : "5A1AE1B3-BB51-4AA9-8B06-853BF9BDD66D",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-5s",
        "state" : "Shutdown",
        "name" : "iPhone 5s"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/97BED1EA-1033-4628-9634-5B06B5D0C003\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/97BED1EA-1033-4628-9634-5B06B5D0C003",
        "udid" : "97BED1EA-1033-4628-9634-5B06B5D0C003",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-6-Plus",
        "state" : "Shutdown",
        "name" : "iPhone 6 Plus"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/274665AD-6F8C-42E2-85D1-C5D961663EEA\/data",
        "dataPathSize" : 805748736,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/274665AD-6F8C-42E2-85D1-C5D961663EEA",
        "udid" : "274665AD-6F8C-42E2-85D1-C5D961663EEA",
        "isAvailable" : true,
        "logPathSize" : 6709248,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-6",
        "state" : "Shutdown",
        "name" : "iPhone 6"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/4BA37798-A816-4993-982E-89FC2CF62B50\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/4BA37798-A816-4993-982E-89FC2CF62B50",
        "udid" : "4BA37798-A816-4993-982E-89FC2CF62B50",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-6s",
        "state" : "Shutdown",
        "name" : "iPhone 6s"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/6A68069F-5A94-40DE-8E66-748A2263C6B6\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/6A68069F-5A94-40DE-8E66-748A2263C6B6",
        "udid" : "6A68069F-5A94-40DE-8E66-748A2263C6B6",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-6s-Plus",
        "state" : "Shutdown",
        "name" : "iPhone 6s Plus"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/280F0A9D-FC26-4C6C-86D6-5D6675271F96\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/280F0A9D-FC26-4C6C-86D6-5D6675271F96",
        "udid" : "280F0A9D-FC26-4C6C-86D6-5D6675271F96",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-SE",
        "state" : "Shutdown",
        "name" : "iPhone SE (1st generation)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/5C3F0CE9-4070-49D8-AC14-C3415F2228C7\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/5C3F0CE9-4070-49D8-AC14-C3415F2228C7",
        "udid" : "5C3F0CE9-4070-49D8-AC14-C3415F2228C7",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-7",
        "state" : "Shutdown",
        "name" : "iPhone 7"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/19EBEB47-C6D8-4022-8389-72336C643167\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/19EBEB47-C6D8-4022-8389-72336C643167",
        "udid" : "19EBEB47-C6D8-4022-8389-72336C643167",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-7-Plus",
        "state" : "Shutdown",
        "name" : "iPhone 7 Plus"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/CA390A8A-0951-4166-B0DE-6923DD23BF22\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/CA390A8A-0951-4166-B0DE-6923DD23BF22",
        "udid" : "CA390A8A-0951-4166-B0DE-6923DD23BF22",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-8",
        "state" : "Shutdown",
        "name" : "iPhone 8"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/FB926175-6FEA-4BD4-8B66-6E8DFC8B6ED5\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/FB926175-6FEA-4BD4-8B66-6E8DFC8B6ED5",
        "udid" : "FB926175-6FEA-4BD4-8B66-6E8DFC8B6ED5",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-8-Plus",
        "state" : "Shutdown",
        "name" : "iPhone 8 Plus"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/19A5F921-F935-458C-BD41-69F75EFBCB52\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/19A5F921-F935-458C-BD41-69F75EFBCB52",
        "udid" : "19A5F921-F935-458C-BD41-69F75EFBCB52",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-X",
        "state" : "Shutdown",
        "name" : "iPhone X"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/F85036ED-E801-4D74-A520-9396E33CEE2E\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/F85036ED-E801-4D74-A520-9396E33CEE2E",
        "udid" : "F85036ED-E801-4D74-A520-9396E33CEE2E",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-XS",
        "state" : "Shutdown",
        "name" : "iPhone Xs"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/8F6429CA-9295-434D-BA17-BA47CBBD6BCE\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/8F6429CA-9295-434D-BA17-BA47CBBD6BCE",
        "udid" : "8F6429CA-9295-434D-BA17-BA47CBBD6BCE",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-XS-Max",
        "state" : "Shutdown",
        "name" : "iPhone Xs Max"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/5DD9BED8-E9E9-4FD8-8573-5B778CEA96A5\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/5DD9BED8-E9E9-4FD8-8573-5B778CEA96A5",
        "udid" : "5DD9BED8-E9E9-4FD8-8573-5B778CEA96A5",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-XR",
        "state" : "Shutdown",
        "name" : "iPhone Xʀ"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/E872CD0A-A49A-4894-B03C-4055DDF4E1D9\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/E872CD0A-A49A-4894-B03C-4055DDF4E1D9",
        "udid" : "E872CD0A-A49A-4894-B03C-4055DDF4E1D9",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Air",
        "state" : "Shutdown",
        "name" : "iPad Air"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/4AE50DAB-5C63-4AFC-AA67-CA234A78349E\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/4AE50DAB-5C63-4AFC-AA67-CA234A78349E",
        "udid" : "4AE50DAB-5C63-4AFC-AA67-CA234A78349E",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Air-2",
        "state" : "Shutdown",
        "name" : "iPad Air 2"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/F22CC3C8-A1FE-456A-BC5F-E4AE3606AC52\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/F22CC3C8-A1FE-456A-BC5F-E4AE3606AC52",
        "udid" : "F22CC3C8-A1FE-456A-BC5F-E4AE3606AC52",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Pro--9-7-inch-",
        "state" : "Shutdown",
        "name" : "iPad Pro (9.7-inch)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/60A58ADE-AF8C-4106-9C19-387E9E18B0AB\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/60A58ADE-AF8C-4106-9C19-387E9E18B0AB",
        "udid" : "60A58ADE-AF8C-4106-9C19-387E9E18B0AB",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Pro",
        "state" : "Shutdown",
        "name" : "iPad Pro (12.9-inch) (1st generation)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/EEA501D5-FEFB-4182-8A30-1F83CE25932F\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/EEA501D5-FEFB-4182-8A30-1F83CE25932F",
        "udid" : "EEA501D5-FEFB-4182-8A30-1F83CE25932F",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad--5th-generation-",
        "state" : "Shutdown",
        "name" : "iPad (5th generation)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/D8E2D8ED-8D97-4F91-88A4-ADEA531C8E3A\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/D8E2D8ED-8D97-4F91-88A4-ADEA531C8E3A",
        "udid" : "D8E2D8ED-8D97-4F91-88A4-ADEA531C8E3A",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Pro--12-9-inch---2nd-generation-",
        "state" : "Shutdown",
        "name" : "iPad Pro (12.9-inch) (2nd generation)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/3378D71B-6F93-44F5-B4DC-7AB790355D96\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/3378D71B-6F93-44F5-B4DC-7AB790355D96",
        "udid" : "3378D71B-6F93-44F5-B4DC-7AB790355D96",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Pro--10-5-inch-",
        "state" : "Shutdown",
        "name" : "iPad Pro (10.5-inch)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/A9A4AC74-F56F-4146-99DA-4997831261D0\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/A9A4AC74-F56F-4146-99DA-4997831261D0",
        "udid" : "A9A4AC74-F56F-4146-99DA-4997831261D0",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad--6th-generation-",
        "state" : "Shutdown",
        "name" : "iPad (6th generation)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/C2F8BF68-6FE1-43D5-BD4D-E06CCAEA498F\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/C2F8BF68-6FE1-43D5-BD4D-E06CCAEA498F",
        "udid" : "C2F8BF68-6FE1-43D5-BD4D-E06CCAEA498F",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Pro--11-inch-",
        "state" : "Shutdown",
        "name" : "iPad Pro (11-inch) (1st generation)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/C23B6A8E-09A7-4BD7-9A68-19417BFEA48E\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/C23B6A8E-09A7-4BD7-9A68-19417BFEA48E",
        "udid" : "C23B6A8E-09A7-4BD7-9A68-19417BFEA48E",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Pro--12-9-inch---3rd-generation-",
        "state" : "Shutdown",
        "name" : "iPad Pro (12.9-inch) (3rd generation)"
      },
      {
        "dataPath" : "\/Users\/magder\/Library\/Developer\/CoreSimulator\/Devices\/396A22F0-C5C8-4CA3-9C71-F0ACF9438C11\/data",
        "dataPathSize" : 18341888,
        "logPath" : "\/Users\/magder\/Library\/Logs\/CoreSimulator\/396A22F0-C5C8-4CA3-9C71-F0ACF9438C11",
        "udid" : "396A22F0-C5C8-4CA3-9C71-F0ACF9438C11",
        "isAvailable" : true,
        "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPad-Air--3rd-generation-",
        "state" : "Shutdown",
        "name" : "iPad Air (3rd generation)"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.tvOS-15-4" : [

    ],
    "com.apple.CoreSimulator.SimRuntime.iOS-15-4" : [

    ]
  }
}
```

Comment (jmagman) [reply lib/web_ui/dev/safari_installation.dart:258]: If you don't want to deal with changing the parsing in this review you could keep the text parsing and still add the version and "available"

```suggestion
        await io.Process.run('xcrun', <String>['simctl', 'list', 'devices', 'iOS $osMajorVersion.osMinorVersion', 'available']);
```

**Review by jmagman — APPROVED** [review]
As a copy and removing dependency this LGTM, changing the `simctl` command is optional bonus (or could be done in another PR).

File: lib/web_ui/dev/safari_installation.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/dev/safari_installation.dart]: That's exactly the kind of improvements we can easily make and test now. But yeah, I'll delay any major fixes to a later time. I accepted `devices iOS major.minor available` though. Thanks!

Comment (jmagman) [reply lib/web_ui/dev/safari_installation.dart]: Well.. that didn't work.
```
Error getting iOS Simulator for iPhone 11 with iOS 13.0.
Try running `felt create` command before running tests.
Unhandled exception:
Exception: Simulator of iPhone 11 is not available for iOS version 13.0
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8816148320707432017/+/u/test:_Run_tests_on_iOS_Safari/stdout

(it totally is available)
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8816148320707432017/+/u/list_simulators__2_/stdout

Comment (jmagman) [reply lib/web_ui/dev/safari_installation.dart]: (this is what it looks like on my machine, which isn't running the same version of Xcode)
```
$ xcrun simctl  list devices "iOS 13.0" available
== Devices ==
-- iOS 12.4 --
-- iOS 13.0 --
    iPhone 8 (C6B00920-CD10-4CD7-902B-6DA1717BA083) (Shutdown)
    iPhone 8 Plus (3AACAD52-1A06-4670-9DE9-D2C98740DAD5) (Shutdown)
    iPhone 11 (D4792554-548B-4F74-9D01-505DC284EED5) (Shutdown)
    iPhone 11 Pro (0E08D4FE-DE75-4879-92B9-B2D8760710AB) (Shutdown)
    iPhone 11 Pro Max (5D79A627-26E1-41E2-BC80-92515ABDA395) (Shutdown)
    iPad Pro (9.7-inch) (AA9559A7-1679-45F1-87CA-20FFA0F0EE1A) (Shutdown)
    iPad Pro (11-inch) (1st generation) (3F474BD8-461C-42A8-8469-A04E8780AB6A) (Shutdown)
    iPad Pro (12.9-inch) (3rd generation) (321C8BDC-993C-45CE-A785-40DF08CEA02C) (Shutdown)
    iPad Air (3rd generation) (2CB4C5C1-0BC9-4D65-88CA-91BB160D5D4D) (Shutdown)
-- iOS 14.0 --
-- iOS 15.4 --
-- tvOS 15.4 --
-- watchOS 8.5 --
```
Not sure why it's failing here, but I guess revert the change I suggested...

Comment (yjbanov) [reply lib/web_ui/dev/safari_installation.dart:258]: Bummer. I'll try again when I have a mac computer where I can debug this.

**Comment by Hixie** [comment]
test-exempt: if this is just moving code between repos; if there's new code though please do test it.

**Comment by skia-gold** [comment]
Gold has detected about 12 new digest(s) on patchset 2.
View them at https://flutter-engine-gold.skia.org/cl/github/32852

**Pull request merged**